_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1030 @@
+              "query_text": "wild west",
+              "model_id": "aVeif4oB5Vm0Tdw8zYO2",
+              "k": 5
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Not only does OpenSearch return documents that match the semantic meaning of `wild west`, but now the documents containing the words related to the wild west theme are scored higher relative to the others:
+
+<details closed markdown="block">
+  <summary>
+    Results
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 26,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 5,
+      "relation": "eq"
+    },
+    "max_score": 0.8744404,
+    "hits": [
+      {
+        "_index": "my-nlp-index",
+        "_id": "5",
+        "_score": 0.8744404,
+        "_source": {
+          "text": "A rodeo cowboy , wearing a cowboy hat , is being thrown off of a wild white horse .",
+          "id": "2691147709.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "3",
+        "_score": 0.5005,
+        "_source": {
+          "text": "People line the stands which advertise Freemont 's orthopedics , a cowboy rides a light brown bucking bronco .",
+          "id": "2664027527.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "4",
+        "_score": 0.5005,
+        "_source": {
+          "text": "A man who is riding a wild horse in the rodeo is very near to falling off .",
+          "id": "4427058951.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "2",
+        "_score": 0.48855463,
+        "_source": {
+          "text": "A wild animal races across an uncut field with a minimal amount of trees.",
+          "id": "1775029934.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "1",
+        "_score": 0.21370724,
+        "_source": {
+          "text": "A West Virginia university women 's basketball team , officials , and a small gathering of fans are in a West Virginia arena .",
+          "id": "4319130149.jpg"
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+Instead of specifying the search pipeline with every request, you can set it as a default search pipeline for the index as follows:
+
+```json
+PUT /my-nlp-index/_settings 
+{
+  "index.search.default_pipeline" : "nlp-search-pipeline"
+}
+```
+{% include copy-curl.html %}
+
+You can now experiment with different weights, normalization techniques, and combination techniques. For more information, see [`normalization_processor`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/) and [`hybrid` query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/) documentation.
+
+### Clean up
+
+After you're done, delete the components you've created from the cluster:
```

Comment:
What's the purpose of the clean-up task?

Reply:
Deleting the model, model group, index, and pipelines.